code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -54,6 +62,12 @@
 	if(istype(O, /obj/item/toy/carpplushie/dehy_carp))
 		var/obj/item/toy/carpplushie/dehy_carp/dehy = O
 		dehy.Swell() // Makes a carp
+	if(!istype(O, /atom/movable/lighting_overlay) && O.has_been_invisible_sprayed)
```

Comment:
This should check for `O.simulated`, not an istype for lighting_overlays... that said, I don't think `reaction_obj` can actually be called for lighting overlays. I hope not, anyways.